src/DrupalConsoleStack.php

```diff
@@ -0,0 +1,698 @@
+    {
+        $this->executable = $pathToDrupalConsole;
+    }
+
+    /**
+     * Pass argument to executable; applies to the next command only.
+     *
+     * @param string $arg
+     *   The argument to pass.
+     *
+     * @return $this
+     */
+    protected function argForNextCommand($arg)
+    {
+        return $this->argsForNextCommand($arg);
+    }
+
+    /**
+     * Pass methods parameters as arguments to executable;
+     * applies to the next command only.
+     *
+     * @param mixed $args,...
+     *   If it's an array, each element is an argument, if not, each parameter to
+     *   this method is an argument.
+     *
+     * @return $this
+     */
+    protected function argsForNextCommand($args)
+    {
+        if (!is_array($args)) {
+            $args = func_get_args();
+        }
+        $this->argumentsForNextCommand .= " " . implode(' ', $args);
+        return $this;
+    }
+
+    /**
+     * Sets the Drupal root directory option.
+     *
+     * @param string $drupalRootDirectory
+     *   The path to the Drupal root directory.
+     *
+     * @return $this
+     */
+    public function drupalRootDirectory($drupalRootDirectory)
+    {
+        $this->printTaskInfo('Drupal root: <info>' . $drupalRootDirectory . '</info>');
+        $this->option('--root', $drupalRootDirectory);
+
+        return $this;
+    }
+
+    /**
+     * Sets the URI of the Drupal site.
+     *
+     * @param string $uri
+     *   URI of the Drupal site to use (for multi-site environments or when
+     *   running on an alternate port).
+     *
+     * @return $this
+     */
+    public function uri($uri)
+    {
+        $this->printTaskInfo('URI: <info>' . $uri . '</info>');
+        $this->option('--uri', $uri);
+
+        return $this;
+    }
+
+    /**
+     * Sets the environment name.
+     *
+     * @param string $environment
+     *   The environment name.
+     *
+     * @return $this
+     */
+    public function environment($environment = 'prod')
+    {
+        $this->printTaskInfo('Environment: <info>' . $environment . '</info>');
+        $this->option('--env ', $environment);
+
+        return $this;
+    }
+
+    /**
+     * Switches off debug mode.
+     *
+     * @return $this
+     */
+    public function noDebug()
+    {
+        $this->option('--no-debug');
+
+        return $this;
+    }
+
+    /**
+     * Sets the verbosity level.
+     *
+     * @param int $level
+     *   One of the DrupalConsoleStack::VERBOSITY_LEVEL_* constants.
+     *
+     * @return $this
+     */
+    public function verbose($level = self::VERBOSITY_LEVEL_NORMAL)
+    {
+        $this->option('--verbose', $level);
+
+        return $this;
+    }
+
+    /**
+     * Sets the site name.
+     *
+     * @param string $siteName
+     *   The site name.
+     *
+     * @return $this
+     */
+    public function siteName($siteName)
+    {
+        $this->argForNextCommand('--site-name=' . escapeshellarg($siteName));
```

Comment:
We don't seem to be consistent in when we escape the argument and when we don't. Maybe we should split up the arguments for argForNextCommand in to $option and $value. Do something like:

```
protected function argForNextCommand($option, $value = NULL)
{
    return $this->argsForNextCommand([[$option => $value]]);
}

protected function argsForNextCommand($args)
{
    foreach ($args as $arg) {
        foreach ($arg as $option => $value) {
            $this->argumentsForNextCommand .= ' --' . $option . (is_null($value) ? '' : '=' . escapeshellarg($value));
        }
    return $this;
}
```